### PR TITLE
[frenemies] Fetch sui balance

### PR DESCRIPTION
**Pull request by Jordan-Mysten**

### CODE REVIEW

File: dapps/frenemies/package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "@mysten/wallet-kit": "workspace:*",
     "@tanstack/react-query": "^4.22.0",
     "@tanstack/react-query-devtools": "^4.22.0",
+    "bignumber.js": "^9.1.1",
```
Comment (damirka) [inline dapps/frenemies/package.json:21]: Are you sure we need this buddy? Won't wrapping into BigInt solve the issue? Besides we use bigint everywhere in our type definitions.

File: dapps/frenemies/src/components/Validators/Balance.tsx [hunk 2]
```diff
@@ -1,18 +1,39 @@
...
+import provider from "../../network/provider";
+
 export function Balance() {
+  const { currentAccount } = useWalletKit();
+  const { data } = useQuery(
+    ["account", "balance"],
+    async () => {
+      const { decimals }= await provider.getCoinMetadata(SUI_TYPE_ARG);
+      const { totalBalance } = await provider.getBalance(currentAccount!, SUI_TYPE_ARG);
+      return new BigNumber(totalBalance).shiftedBy(-1 * decimals).toFormat();
```
Comment (damirka) [inline dapps/frenemies/src/components/Validators/Balance.tsx:17]: I think we have a SUI formatting function in Capys, I can pull it here. Not to introduce unnecessary dependency (like we did with `bcs` library).

**Review by damirka — APPROVED** [review]
Approving to unblock. But I think we can avoid using BN.js here.

File: dapps/frenemies/src/components/Validators/Balance.tsx (hunk 2 above)
Comment (Jordan-Mysten) [reply dapps/frenemies/src/components/Validators/Balance.tsx:17]: I'd honestly rather keep this here, I think it's fine to have a dep for this.

**Pull request merged**